### PR TITLE
Simplify proofreader installation on VMs

### PR DESCRIPTION
**Pull request by giorgiosironi**
Abandon the Composer installation and install into a Git repository like for many other applications. From there it's a simple symlink into a folder on the PATH to use `proofreader`.

### CODE REVIEW

File: elife/proofreader-php.sls [hunk 1]
```diff
@@ -1,6 +1,59 @@
...
+        - recurse:
+            - user
+            - group
+        - require:
+            - builder: proofreader-php-repository
+
     cmd.run:
-        - name: composer global --no-interaction require --update-with-dependencies elife/proofreader-php='dev-master#83901bb097b39b0df4ddc9d0e841b685977131ab'
+        - name: |
+            composer install
```
Comment (thewilkybarkid) [inline elife/proofreader-php.sls]: Since we want speed here, `--no-suggest --classmap-authoritative --no-dev`.

File: elife/proofreader-php.sls [hunk 2]
```diff
@@ -1,6 +1,59 @@
...
         - user: {{ pillar.elife.deploy_user.username }}
+        - cwd: /srv/proofreader-php
```
Comment (thewilkybarkid) [inline elife/proofreader-php.sls]: I'd always assumed `srv` = serve = web apps.

**Comment by thewilkybarkid** [comment]
Should we not make use of https://github.com/elifesciences/proofreader-php/pull/5?

File: elife/proofreader-php.sls (hunk 1 above)
Comment (giorgiosironi) [reply elife/proofreader-php.sls]: Yes, this should never be used for development of `proofreader`.

File: elife/proofreader-php.sls (hunk 2 above)
Comment (giorgiosironi) [reply elife/proofreader-php.sls]: We install practically everything created after 2016 there, useful to sometimes mount `/srv` on an external volume and in general treating it as "where the `elife` user puts stuff"

**Comment by giorgiosironi** [comment]
> Should we not make use of elifesciences/proofreader-php#5?

It would required `elife.docker` on all `*--ci` instances, which has implications for disk space usage and provisioning time. I'd move to it when each project moves towards containers (and hence its `--ci` instance disappears) while this should be backward compatible with its current `composer global` installation.

**Pull request merged**